Adds docs on 404 page (#1213)
rstudio/bookdown: inst/examples/02-components.Rmd

Again, you will see a live app if you are reading an HTML version of this book, and a static screenshot if you are reading other types of formats. The above Shiny app was created using the **miniUI** package [@R-miniUI], which provides layout functions that are particularly nice for Shiny apps on small screens. If you use normal Shiny layout functions, you are likely to see vertical and/or horizontal scrollbars in the iframes because the page size is too big to fit in an iframe. When the default width of the iframe is too small, you may use the chunk option `out.width` to change it. For the height of the iframe, use the `height` argument of `include_url()`/`include_app()`.

Shiny apps may take even longer to load than usual URLs. You may want to use a conservative value for the `delay` option, e.g., 10. Needless to say, `include_url()` and `include_app()` require a working Internet connection, unless you have previously cached the chunk (but web pages inside iframes still will not work without an Internet connection).

## HTML 404 pages {#html-404}

> 

Reviewer: I like the fact that the 404 section in Quarto doc on this page (https://quarto.org/docs/websites/website-navigation.html)
mentions two things: 

* A brief mention about 404 page are for 404 error with a link to know more. 
* Mention that service need to support it (Netlify, Github page)

I'll add sugestion below based on that.

> By default, both the `gitbook()` (Section \@ref(gitbook-style)) and `bs4_book()` (Section \@ref(bs4-book)) HTML output formats include a 404 page for readers who try to access a page in your book that cannot be found. The default 404.html is created using simple content (a header, and a body of 2 paragraphs). 

Reviewer: ```suggestion
If a readers try to try to access a page in your book that cannot be found (e.g a link to a page that no more exists), a browser will display a [404 error](https://en.wikipedia.org/wiki/HTTP_404) as it can’t find the requested web page. This 404 error is displayed on a 404 page that each browser has a default for. You may want to customized this page to be in the style of your HTML book. 
Most web serving platforms (e.g. Netlify, Github Pages, or Gitlab Pages) will use a file named `404.html` in the root of your website as a custom error page if you provide it. 

With HTML book format, **bookdown** creates a default `404.html`  using simple content (a header, and a body of 2 paragraphs).  
```

```{r 404-page, fig.alt='Screenshot of an example 404 page for an HTML book built with bs4_book.', fig.align='center', echo=FALSE}
knitr::include_graphics('images/404.png')
```

> To customize the 404 page instead of using the default, you may add either a `_404.Rmd` or a `_404.md` file to your project root. If either file is found when you render the book, the content will be rendered as an `html_fragment` and will be included as the body of the 404 page. This means that the overall structure of your book's website (including navbar, footer, sidebars, etc.) and the CSS styling will be preserved on the 404 page.

Reviewer: ```suggestion
To customize the 404 page instead of using the default, you may add either a `_404.Rmd` or a `_404.md` file to your project root. If either file is found when you render the book, the content will be rendered as an `html_fragment()` and will be included as the body of the 404 page. This means that the overall structure of your book's website (including navbar, footer, sidebars, etc.) and the CSS styling will be preserved on the 404 page.
```

> If a 404.html file already exists in the main repo at the root level with the `.Rmd` files, then **bookdown** will leave that file as is and will not overwrite it.

Reviewer: I want to add this to explain
```suggestion
If a 404.html file already exists in the main repo at the root level with the `.Rmd` files, then **bookdown** will leave that file as is and will not overwrite it. This is because we assume you already have a mechanism in place in your publishing workflow to use this custom `404.html` you had setup.
```

but maybe unnecessary

Again, you will see a live app if you are reading an HTML version of this book, and a static screenshot if you are reading other types of formats. The above Shiny app was created using the **miniUI** package [@R-miniUI], which provides layout functions that are particularly nice for Shiny apps on small screens. If you use normal Shiny layout functions, you are likely to see vertical and/or horizontal scrollbars in the iframes because the page size is too big to fit in an iframe. When the default width of the iframe is too small, you may use the chunk option `out.width` to change it. For the height of the iframe, use the `height` argument of `include_url()`/`include_app()`.

Shiny apps may take even longer to load than usual URLs. You may want to use a conservative value for the `delay` option, e.g., 10. Needless to say, `include_url()` and `include_app()` require a working Internet connection, unless you have previously cached the chunk (but web pages inside iframes still will not work without an Internet connection).

## HTML 404 pages {#html-404}

> If a reader tries to access a page in your book that cannot be found, a browser will display a [404 error](https://en.wikipedia.org/wiki/HTTP_404) as it cannot find the requested web page. This 404 error is displayed on a 404 page. Each browser has a default for a 404 page. However, most web serving platforms like Netlify, Github Pages, and Gitlab Pages will use a file named `404.html` in the root of your website as a custom error page, if you provide it. 

Reviewer 2: The default 404 page comes from the server, not browser:

```suggestion
If a reader tries to access a page in your book that cannot be found, a browser will display a [404 error](https://en.wikipedia.org/wiki/HTTP_404) as it cannot find the requested web page. This 404 error is displayed on a 404 page. Each web server has a default for a 404 page. However, most web serving platforms like Netlify, Github Pages, and Gitlab Pages will use a file named `404.html` in the root of your website as a custom error page, if you provide it. 
```

Reviewer: Yes server usually have défaut 404. It is more common. I am quite certain that browser can also have there default 404 pages, but I guess it could have been mainly IE on windows.

For all HTML book formats, **bookdown** creates a custom `404.html`  in your output directory using simple content (a header, and a body of 2 paragraphs); see Figure \@ref(fig:404-page).  

```{r 404-page, fig.alt="Screenshot showing an online bs4_book with the sidebar, footer, and CSS styling in place. The text reads: Page not found. The page you requested cannot be found (perhaps it was moved or renamed). You may want to try searching to find the page's new location, or use the table of contents to find the page you are looking for.", fig.align='center', echo=FALSE, fig.cap= "Screenshot of an example 404 page for an HTML book built with bs4_book."}
knitr::include_graphics('images/404.png')
```

As you can see, this 404 page is embedded within the book so that readers can quickly find their way back to the book's content. The overall structure of the book's website (including navbar, footer, sidebars, etc.) and the CSS styling are preserved on the 404 page.

> To customize the 404 page instead of using the one **bookdown** provides, you may add either a `_404.Rmd` or a `_404.md` file to your project root. If either file is found when you render the book, the content will be rendered as an `html_fragment()` and will be included as the body of the 404 page embedded within the book structure.

Reviewer 2: I suggest we omit the implementation detail `html_fragment()`:

```suggestion
To customize the 404 page instead of using the one **bookdown** provides, you may add either a `_404.Rmd` or a `_404.md` file to your project root. If either file is found when you render the book, the content will be rendered and included as the body of the 404 page embedded within the book structure.
```
